users/config: Update folder docs (#525)
syncthing/docs: users/config.rst

        <markerName>.stfolder</markerName>
        <copyOwnershipFromParent>false</copyOwnershipFromParent>
        <modTimeWindowS>0</modTimeWindowS>
        <maxConcurrentWrites>2</maxConcurrentWrites>
        <disableFsync>false</disableFsync>
        <blockPullOrder>standard</blockPullOrder>

>         <copyRangeMethod>standard</copyRangeMethod>

Reviewer: The same 4 lines should probably be added to the folder element under https://github.com/syncthing/docs/blob/new-docs/users/config.rst#config-file-format (i.e. between the lines 71 and 72).

    Name of a directory or file in the folder root to be used as
    :ref:`marker-faq`. Default is ".stfolder".

copyOwnershipFromParent
    On Unix systems, tries to copy file/folder ownership from the parent directory (the directory it's located in).
    Requires running Syncthing as privileged user, or granting it additional capabilities (e.g. CAP_CHOWN on Linux).

modTimeWindowS

>     Allowed modification timestamp difference when comparing local and remote file for equivalence.

Reviewer 2: ```suggestion
    Allowed modification timestamp difference when comparing files for equivalence.
```

Also applies in the scanner (local to local).

    To be used on systems that have unstable modification timestamps, that might change after being observed after
    the last write operation. Used in Android only.

maxConcurrentWrites
    Maximum number of concurrent write operations while syncing. Defaults to 2. Increasing this might increase or
    decrease disk performance, depending on the underlying storage.

disableFsync

>     Disables committing file operations to disk before recording them in the database.

Reviewer 2: I'd discourage usage even more, something like this:

```suggestion
    Do not enable this option unless you know exactly what you are doing.
    Disables committing file operations to disk before recording them in the database.
```

And or use the "warning-thingy" (just discovered below).

>     Can lead to data corruption if disabled.

Reviewer 2: Potentially confusing, given the `disableFsync` setting being enabled is dangerous (which disables fsync). 

```suggestion
    Disabling fsync can lead to data corruption.
```

blockPullOrder
    Order in which parts of a file are downloaded. This option controls how quickly the different parts of the
    file spread between the connected devices, at the cost of causing strain on the storage.

    Available options:

    standard (default):

>         The file is split into N equal continuous part sequences, where N is the number of connected

Reviewer 2: ```suggestion
        The file is split into N equal continuous sequences, where N is the number of connected
```

Author: I think it's imporant to leave part here, because file is split into parts, so it's important to know its sequences of parts.

Reviewer 2: Ah, now I get what distinction you want to stress here. I'd use "blocks" then e.g.

```suggestion
        The blocks of the file are split into N equal continuous sequences, where N is the number of connected
```

>         devices, each device starts downloading it's own sequence, after which it picks other devices

Reviewer 2: ```suggestion
        devices. Each device starts downloading it's own sequence, after which it picks other devices
```

        sequence at random. Provides acceptable data distribution, minimal spinning disk strain.

    random:
        File parts are downloaded in a random order. Provides great data distribution, but very taxing on
        spinning disk drives

    inOrder:
        File parts are downloaded sequentially, from start to finish. Spinning disk drive friendly, but provides
        no improvements to data distribution.

copyRangeMethod

>     Provides an choice of method for copying data between files. This can be used to optimise copies on network

Reviewer 2: ```suggestion
    Provides a choice of method for copying data between files. This can be used to optimise copies on network
```

        BTRFS_IOC_CLONE_RANGE, which was used to provide copy-on-write semantics to BTRFS filesystems since Linux 2.6.29.
        Some network filesystems might use this to perform server-side copies. Will fail if not supported by the
        underlying filesystem.

        Tested on: BTRFS
        Not available on: Windows, Solaris, Darwin (OS X)

    sendfile
        Uses sendfile syscall, which performs in-kernel copy, avoiding having to copy the data into application memory.

        Tested on: BTRFS, XFS
        Not available on: Windows, Darwin (OS X)

    duplicate_extents
        Uses Windows Block Cloning via FSCTL_DUPLICATE_EXTENTS_TO_FILE, which provides copy-on-write semantics to clone
        the data. Requires Windows Server 2016 or later, and a compatible filesystem (ReFS, SMB 3.1.1, CsvFS). Will
        fail if not supported by the underlying filesystem.

        .. warning::
            Completely untested, use at your own risk.

        Available on: Windows

    all
        Tries all of the copy methods in the following order: ioctl, copy_file_range, sendfile, duplicate_extents,
        standard.

        Available on: All platforms

        .. warning::

>             Not recommended on Windows as it has not been tested, might be very costly on all other platforms.

Reviewer 2: This could be read as implying it's been tested, regardless of the initial warning. I'd reword to

```suggestion
            Not recommended as it has not been tested. Also might be very costly on all platforms but Windows.
```

Author: This was tested on non-windows to some extent.
